code-challenges.js

```diff
@@ -23,11 +43,47 @@ var middleLetters1 = "hello"
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
+var middleLetters1 = "hello"
+// Expected output: “l”
+
+var middleLetters2 = "rhinoceros"
+// Expected output: “oc”
+
+function middle(str) {
+  var position, length;
+  if ((str.length % 2) == 0) {
+    position = str.length / 2 - 1;
+    length = 2;
+  } else {
+    position = (str.length - 1) / 2;
+    length = 1;
+  }
+
+  return str.substring(position, position + length);
+}
+console.log(middle("hello"))
+console.log(middle("rhinoceros"))
 
 
 
 // --------------------4) Create a class to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
 
+let Sphere = class {
+	constructor(radius) {
+		this.radius = radius;
+	}
+	get area() {
+		return this.calcArea()
+	}
+
+	calcArea() {
+		return ((this.radius * this.radius) * Math.PI * 4)
+	}
```

Comment:
if the sole purpose of `area()` is to call another function that's handling your arithmetic, you can probably just delete `area()` method